MakingABackend.MD

```diff
@@ -194,4 +194,423 @@ class AwsJobExecutionActor(override val jobDescriptor: BackendJobDescriptor,
+                           (implicit ec: ExecutionContext): ExecutionHandle = {
+  try {
+    if (isSuccess(status)) {
+
+      lazy val stderrLength: Long = File(remoteStrErrPath).size
+      lazy val returnCode: Try[Int] = returnCodeContents.map(_.trim.toInt)
+      status match {
+        case _ if failOnStderr && stderrLength.intValue > 0 =>
+          // returnCode will be None if it couldn't be downloaded/parsed, which will yield a null in the DB
+          FailedNonRetryableExecutionHandle(new RuntimeException(
+            s"execution failed: stderr has length $stderrLength"), returnCode.toOption)
+        case _ if returnCodeContents.isFailure =>
+          val exception = returnCode.failed.get
+          jobLogger.warn(s"could not download return code file, retrying", exception)
+          // Return handle to try again.
+          handle
+        case _ if returnCode.isFailure =>
+          FailedNonRetryableExecutionHandle(new RuntimeException(
+            s"execution failed: could not parse return code as integer: ${returnCodeContents.get}"))
+        case _ if !continueOnReturnCode.continueFor(returnCode.get) =>
+          val badReturnCodeMessage = s"Call ${jobDescriptor.key}: return code was ${returnCode.getOrElse("(none)")}"
+          FailedNonRetryableExecutionHandle(new RuntimeException(badReturnCodeMessage), returnCode.toOption)
+        case _ =>
+          handleSuccess(status)
+      }
+
+    } else {
+      handleFailure(status)
+    }
+  } catch {
+    case e: Exception =>
+      jobLogger.warn("Caught exception processing job result, retrying", e)
+      // Return the original handle to try again.
+      handle
+  }
+}
+
+private def commandLine: String = jobDescriptor.call.instantiateCommandLine(
+  commandLineInputs, commandLineFunctions, commandLineValueMapper).get
+
+private lazy val tag = s"${this.getClass.getSimpleName} [UUID(${workflowId.shortString}):${jobDescriptor.key.tag}]"
+```
+
+### Adding our specifics async implementation:
+
+- There are a few types used above, mainly concerned with the pending execution handle, the job id, and the run status. We'll first start with filling in the type aliases:
+
+```scala
+type BackendJobId = AwsJobId
+type BackendRunInfo = Any
+type BackendRunStatus = AwsRunStatus
+type BackendPendingExecutionHandle = PendingExecutionHandle[BackendJobId, BackendRunInfo, BackendRunStatus]
+```
+
+- Next we added basic implementations of the file accessors. These will be will be filled in during a future date.
+
+```scala
+private def commandLineInputs: CallInputs = Map.empty
+
+private def commandLineFunctions: WdlFunctions[WdlValue] = OnlyPureFunctions
+
+private def commandLineValueMapper: WdlValue => WdlValue = identity
+
+private def remoteStrErrPath: Path = ???
+
+private lazy val returnCodeContents: Try[String] = Try("0")
+```
+
+- Then we added basic implementations of the standard runtime attributes. Again, these will be will be filled in during a future date.
```

Comment:
Not to judge if that's your style... but I think `at a future date` might be less ambiguous than `during a future date` :)